lang_GENERIC_base/AST_generic.ml

```diff
@@ -1349,6 +1362,8 @@ and or_type_element =
   | OrEnum of ident * expr option
   (* Java? *)
   | OrUnion of ident * type_
+  (* Rust *)
+  | OrEnumStruct of ident * field list bracket
```

Comment:
OCaml has also tagged union, and C as unions, so we should not need OrEnumStruct I think.

The best is probably to start writing Parse_rust_tree_sitter.ml, put some todo() for constructs you don't find can be handled by AST_generic.ml, and then we can extend AST_generic.ml as needed.